.github/workflows/ci-baseline.yml

```diff
@@ -28,75 +28,126 @@ jobs:
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
+          - image: fedora-34-clang_12-eigen_3.3.7-gtest
+            build_type: Release
+            cmake_extra_args: '-D CMAKE_CXX_STANDARD=20 -D PRESSIO_ENABLE_CXX20=ON'
 
-          #- { os: ubuntu-latest, image: intel_oneapi-eigen_3.3.7-gtest, mode: Release }
-          #- { os: ubuntu-latest, image: intel_oneapi-eigen_3.3.7-gtest, mode: Debug }
-
-    runs-on: ${{ matrix.config.os }}
-    container: pressiomodelreduction/${{ matrix.config.image }}
+    runs-on: ubuntu-latest
+    container: pressiomodelreduction/${{ matrix.image }}
     env:
-      num_cpus: 2
+      num_cpus: 2 # $(cat /proc/cpuinfo | grep processor -c)
       eigen_inc_dir: /usr/local/eigen/install/include/eigen3
       gtest_dir: /usr/local/gtest/install
-      small: small
-      medium: medium
-      large: large
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+
+      - name: Checkout pressio examples
+        uses: actions/checkout@v4
+        with:
+          repository: Pressio/pressio-templates-usage-as-library
+          path: examples
+
+      - name: Example - include only
+        working-directory: examples
+        run: |
+          $CXX \
+            -DPRESSIO_ENABLE_TPL_EIGEN \
+            -I$eigen_inc_dir           \
+            -I../include            \
+            -c main.cpp                \
+          && $CXX -o pressio-example main.o \
+          && ./pressio-example
+
+      - name: Example - include only with embedded definitions
+        working-directory: examples
+        run: |
+          $CXX \
+            -I$eigen_inc_dir           \
+            -I../include               \
+            -c main_with_defines.cpp   \
+          && $CXX -o pressio-example main.o \
+          && ./pressio-example
 
-      - name: configure
+      - name: Configure
         run: |
           cmake -B builddir \
-            -D CMAKE_BUILD_TYPE:STRING=${{ matrix.config.mode }} \
-            -D CMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+            -D CMAKE_BUILD_TYPE:STRING=${{ matrix.build_type }} \
             -D CMAKE_C_COMPILER=$CC \
             -D CMAKE_CXX_COMPILER=$CXX \
-            -D PRESSIO_ENABLE_TPL_EIGEN:BOOL=ON \
-            -D EIGEN_INCLUDE_DIR=$eigen_inc_dir \
+            -D CMAKE_CXX_FLAGS='' \
+            -D CMAKE_INSTALL_PREFIX:PATH=../install \
+            -D CMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+            ${{ matrix.cmake_extra_args }} \
             -D GTEST_ROOT=$gtest_dir \
             -D PRESSIO_ENABLE_DEBUG_PRINT=ON \
             -D PRESSIO_ENABLE_TESTS:BOOL=ON \
-            -D CMAKE_INSTALL_PREFIX:PATH=../install \
-            -D CMAKE_CXX_FLAGS=''
+            -D PRESSIO_ENABLE_TPL_EIGEN:BOOL=ON \
+            -D EIGEN_INCLUDE_DIR=$eigen_inc_dir
+
+      - name: Build
+        run: |
+          cmake --build builddir --target install
```

Comment:
if i understand correctly, here and above you configure and then build/install and then use the installation to test the example usages below AND also use those installed headers to build tests right?

Reply:
> you configure and then build/install and then use the installation to test the example usages below
Correct.
> AND also use those installed headers to build tests right?
Nope, the tests are build in line 114 too. `cmake --build builddir --target install` builds `all` (which includes tests) and then does the installation.